app/grandchallenge/core/utils/query.py

```diff
@@ -8,3 +11,11 @@ def index(queryset, obj):
             return index
 
     return -1
+
+
+def set_seed(seed):
+    # Note: this only works for postgres, if we ever switch dbs, this
+    # may need changing.
+    cursor = connection.cursor()
+    cursor.execute(f"SELECT setseed({seed});")
```

Comment:
Could you explain how this works? From the docs this looks like it sets the seed for the current session. Am I right in thinking this only affects the current request? There's no way that some other worker could call this?

Reply:
It only effects the current request and only the very first subsequent query.